### PR TITLE
Remove extraneous SFPSWAP NOPs.

### PR DESCRIPTION
**Pull request by jasondavies**
### Ticket

tenstorrent/tt-llk#102

### Problem description

SFPSWAP doesn't need to be followed by SFPNOP.  See tenstorrent/tt-llk#102 for details.

### What's changed

- Removed nnecessary SFPNOPs.
- `unary_max_min` optimised slightly by moving repeated immediate loads out of the loop.
- Removed unused include.

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by nvelickovicTT — APPROVED** [review]

**Comment by rtawfik01** [comment]
Were some tests done using this change to prove this does not causes non-determinism? Essentially I would like a test to be looped a few hundred times and achieve the same PCC

**Comment by jasondavies** [comment]
> Were some tests done using this change to prove this does not causes non-determinism? Essentially I would like a test to be looped a few hundred times and achieve the same PCC

I'd be happy to run a quick check on this particular change {unary,binary} {min,max}.

So far quite a lot of extensive testing has been done e.g. I'm using SFPSWAP without NOP in my GCD implementation, which has been tested extensively over the past few weeks.

There's also the [discussion](https://github.com/tenstorrent/tt-llk/issues/102#issuecomment-2807602522) on my original tt-llk issue, which proves experimentally that the cycle count matches the theory:

> These numbers are entirely consistent with the theory that SFPSWAP takes two cycles to execute, unless it is followed by SFPNOP, in which case it takes one and the SFPNOP takes one. Or, in other words, that a one cycle stall is automatically inserted after SFPSWAP if it isn't followed by SFPNOP.

**Comment by jasondavies** [comment]
I ran the following on Wormhole with no issues:

```python
import ttnn
import torch
from random import randint

device = ttnn.open_device(device_id=0)
dtype = ttnn.float32

rand_min = -(2**31)
rand_max = 2**31
size = (1 * 1024, 8 * 1024)

def test(device, f, unary):
    print(f, unary)
    torch_f = getattr(torch, f)
    ttnn_f = getattr(ttnn, f)

    unary_value = float(randint(rand_min, rand_max))
    a_torch = torch.randint(rand_min, rand_max, size).float()
    b_torch = (torch.full(size, unary_value) if unary else torch.randint(rand_min, rand_max, size)).float()
    expected = torch_f(a_torch, b_torch)

    a = ttnn.from_torch(a_torch, dtype=dtype, layout=ttnn.TILE_LAYOUT, device=device)
    b = unary_value if unary else ttnn.from_torch(b_torch, dtype=dtype, layout=ttnn.TILE_LAYOUT, device=device)
    c = ttnn_f(a, b)

    c_torch = ttnn.to_torch(c)

    assert torch.equal(expected, c_torch)

for unary in [False, True]:
    for i in range(100):
        test(device, "minimum", unary)
        test(device, "maximum", unary)
```

**Comment by rtawfik01** [comment]
The problem with issues like these were timing is impacted, we have seen before that non-deterministic behavior occurs. Please ensure that this change has not introduced any timing non-determinism. All 100 runs (I think you should run for even more, maybe 1000), should all have the exact same result as the first run, using the same seed

**Comment by jasondavies** [comment]
> The problem with issues like these were timing is impacted

Timing should not be impacted.  We've discovered that on Wormhole, SFPSWAP is special and always takes 2 cycles: if there is a NOP, then latency(SFPSWAP, NOP) = 2, otherwise it will automatically insert a stall so that the latency is still 2 cycles before the next instruction.

**Comment by jasondavies** [comment]
The following check runs the 4 operations 1000 times on the same 8M dataset, checking for consistency with PyTorch (all passed):

```python
import ttnn
import torch
from random import randint

device = ttnn.open_device(device_id=0)
dtype = ttnn.float32

rand_min = -(2**31)
rand_max = 2**31
size = (1 * 1024, 8 * 1024)

def test(device, f, unary):
    print(f, unary)
    torch_f = getattr(torch, f)
    ttnn_f = getattr(ttnn, f)

    unary_value = float(randint(rand_min, rand_max))
    a_torch = torch.randint(rand_min, rand_max, size).float()
    b_torch = (torch.full(size, unary_value) if unary else torch.randint(rand_min, rand_max, size)).float()
    expected = torch_f(a_torch, b_torch)

    a = ttnn.from_torch(a_torch, dtype=dtype, layout=ttnn.TILE_LAYOUT, device=device)
    b = unary_value if unary else ttnn.from_torch(b_torch, dtype=dtype, layout=ttnn.TILE_LAYOUT, device=device)

    for i in range(1000):
        c = ttnn_f(a, b)
        c_torch = ttnn.to_torch(c)
        assert torch.equal(expected, c_torch)


for unary in [False, True]:
    test(device, "minimum", unary)
    test(device, "maximum", unary)
```

**Comment by jasondavies** [comment]
Although checking data consistency is useful for regression testing, I think the strongest experimental proof is the [elapsed cycle count test](https://github.com/tenstorrent/tt-llk/issues/102#issuecomment-2807602522), which proves that a stall is inserted following SFPSWAP if the NOP is removed.
